File: 프로그래머스/lv3/43238. 입국심사/입국심사.py
def solution(n, times):
    left = 1
    right = max(times)*n
    answer = 0
    while left <= right:
        mid = (left+ right)//2
        
        cnt = 0
        # mid 시간 동안 각 심사관이 심사할 수 있는 사람의 수
        for time in times:
            cnt += mid//time
        # 최소시간을 구해라
        if cnt >= n:
            answer = mid 
            right = mid - 1
        else:
            left = mid + 1
        
    
    return answer

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let right = Math.max(...times) * n
    let left = 1
    
    while(left <= right){
        const mid = Math.floor((left + right) / 2)
        
        let people = 0
        for(const costEachOne of times)
            people += Math.floor(mid / costEachOne)
        
        if(people >= n)
            right = mid-1
        else 
            left = mid+1
    }
    return left
}